BLU: New spells added in 6.45

Hugmeir opened the pull request:
The only change to existing spells:

    * Magic Hammer shares a cooldown with the newly-added Candy Cane
    * The Rose of Destruction & Chelonian Gate, which shared a
      cooldown, now also share it with the newly added Ruby Dynamics

Since those changes would only matter for logs generated after 6.45, I think it's safe to just add these without messing with the layers.

NOTE: Winged Reprobation is *weird* and it's actual cooldown is not going to be properly tracked yet.  That will be handled in a future commit.

Hugmeir committed:
BLU: New spells added in 6.45

The only change to existing spells:

    * Magic Hammer shares a cooldown with the newly-added Candy Cane
    * The Rose of Destruction & Chelonian Gate, which shared a
      cooldown, now also share it with the newly added Ruby Dynamics

Since those changes would only matter for logs generated after 6.45,
I think it's safe to just add these without messing with the layers.

NOTE: Winged Reprobation is *weird* and it's actual cooldown is
not going to be properly tracked yet.  That will be handled in
a future commit.

AkairyuGestalter commented on src/data/ACTIONS/root/BLU.ts:
```diff
@@ -14,6 +14,8 @@ export const BLU_COOLDOWN_GROUP = {
 	NIGHTBLOOM: 23290,
 	THE_ROSE_OF_DESTRUCTION: 23275,
 	MATRA_MAGIC: 23285,
+	MAGIC_HAMMER: 18305,
```
Minor nitpick I didn't pick up on before. Can we call this const BLU_COOLDOWN_GROUPS to better reflect what it's defining?

Hugmeir replied:
Done!

AkairyuGestalter commented on src/data/ACTIONS/root/BLU.ts:
```diff
@@ -1008,4 +1011,196 @@ export const BLU = ensureActions({
+		name: 'Goblin Punch',
+		icon: 'https://xivapi.com/i/003000/003358.png',
+		onGcd: true,
+		gcdRecast: 2500,
```
Should we go ahead and tag the GCDs (or otherwise? BLU's weird) with the speedAttribute: Attributes.SPELL_SPEED property to make sure the timeline display respects the player's speed?

Hugmeir replied:
I think this is a good idea, though I'd rather do it in a separate PR since I'll need to go and check every spell, since SOME of our oGCDs are also affected by SpS.

(also, to be honest... I play with zero SpS both as BLU and RDM so I didn't even realize this was an option)

Hugmeir replied:
Well, that actually only took a couple of minutes to verify in-game, I didn't remember you could just see the tooltip of spells in the spell book without slotting them into your actions.  PR now includes `speedAttribute: Attributes.SPELL_SPEED` for everything that's affected, including three weird edge cases (White Death and Divine Cataract are affected, Phantom Flurry's kick is not)

AkairyuGestalter reviewed (commented):
Two comments not going to block on either if you'd rather not change 'em though. It looks like they might've double-posted though... I'll try to clean that up... [edit] Nope, I guess they actually only showed once after I refreshed the page .-.

Hugmeir committed:
BLU: BLU_COOLDOWN_GROUP => BLU_COOLDOWN_GROUPS

Hugmeir committed:
BLU Actions: Mark all the actions that are affected by SpS

Which strangely includes our raid buffs and mitigation.

Because I keep being surprised by this: Sps does NOT affect
the kick GCD from Phantom Flurry, so that is always going
to be a 2.5s GCD, regardless of spell speed.

AkairyuGestalter reviewed (approved):
Looks good to me!
